desc/objectives/_geometry.py

```diff
@@ -750,24 +771,113 @@ def compute(self, equil_params, surface_params=None, constants=None):
             transforms=constants["equil_transforms"],
             profiles=constants["equil_profiles"],
         )
-        plasma_coords = rpz2xyz(jnp.array([data["R"], data["phi"], data["Z"]]).T)
+        plasma_coords_rpz = jnp.array([data["R"], data["phi"], data["Z"]]).T
+        plasma_coords = rpz2xyz(plasma_coords_rpz)
         if self._surface_fixed:
-            surface_coords = constants["surface_coords"]
+            data_surf = constants["data_surf"]
         else:
-            surface_coords = compute_fun(
+            data_surf = compute_fun(
                 self._surface,
                 self._surface_data_keys,
                 params=surface_params,
                 transforms=constants["surface_transforms"],
                 profiles={},
                 basis="xyz",
-            )["x"]
-        d = safenorm(plasma_coords[:, None, :] - surface_coords[None, :, :], axis=-1)
+            )
+
+        surface_coords = data_surf["x"]
+        diff_vec = plasma_coords[:, None, :] - surface_coords[None, :, :]
+        d = safenorm(diff_vec, axis=-1)
+
+        point_signs = jnp.ones(surface_coords.shape[0])
+        if self._use_signed_distance:
+            surface_coords_rpz = xyz2rpz(surface_coords)
+
+            def _find_angle_vec(R, Z, Rsurf, Zsurf):
```

Comment:
Can this algorithm be used to calculate the distance between a curve and a surface?

Reply:
yes if at same zeta

Reply:
Could we spin this out into it's own utility function? Because we might have use for this other applications (for example, checking if a point is inside the plasma boundary for field line tracing)